pac4j-saml/src/main/java/org/pac4j/saml/crypto/KeyStoreCredentialProvider.java

```diff
@@ -114,12 +148,17 @@ private KeyStore loadKeyStore(final InputStream inputStream, final String storeP
         }
     }
 
-    private String getPrivateKeyAlias(final KeyStore keyStore) {
+	private String getPrivateKeyAlias(final KeyStore keyStore, final String keyStoreAlias) {
         try {
             final Enumeration<String> aliases = keyStore.aliases();
-            if (aliases.hasMoreElements()) {
-                return aliases.nextElement();
-            }
+			while (aliases.hasMoreElements()) {
+	            String currentAlias = aliases.nextElement();
+				if (keyStoreAlias == null) {
```

Comment:
Previously if the keystore had more than one alias you just picked the first one. I have a scenario where my keystore contains more than one alias and the one i need for saml isn't always the first one so if you specify which alias then it uses the correct one.